cmd/thanos/receive.go

```diff
@@ -315,11 +315,17 @@ func runReceive(
 		// In the single-node case, which has no configuration
 		// watcher, we close the chan ourselves.
 		updates := make(chan receive.Hashring, 1)
+
 		if cw != nil {
+			// Before scheduling a job, check if hashring config is valid.
```

Comment:
Scheduling which job? To clarify maybe consider "Check the hashring configurat on before running the watcher."